openhands/controller/agent_controller.py

```diff
@@ -351,9 +354,22 @@ async def _step(self) -> None:
             return
 
         if self._pending_action:
-            logger.debug(
-                f'[Agent Controller {self.id}] waiting for pending action: {self._pending_action}'
-            )
+            current_pending_action = self._pending_action
+
+            # New: Check if the pending action has changed
+            if (
+                not hasattr(self, '_last_pending_action')
+                or self._last_pending_action != current_pending_action
+            ):
+                logger.debug(
+                    f'[Agent Controller {self.id}] waiting for pending action...'
+                )
```

Comment:
TBH I would rather remove this completely. A developer can always add it when they need this exact info, not like we don't add plenty of extra logging when we have a particular issue to debug.

Reply:
You mean the whole "every second" output block? It's only with debug anyways?
But personally, I wouldn't mind having it removed as well.

Reply:
Yes. I think it's just a bad place to put a logger.debug() at all. We don't need it in the vast majority of executions...

Reply:
Agreed, I'll remove that block with the next commit.